server/issue.go

```diff
@@ -5,47 +5,41 @@ package server
 
 import (
 	"context"
+	"errors"
+	"fmt"
 	"strings"
 
 	"github.com/mattermost/mattermost-mattermod/model"
 	"github.com/mattermost/mattermost-server/v5/mlog"
 )
 
-func (s *Server) handleIssueEvent(ctx context.Context, event *PullRequestEvent) {
+func (s *Server) handleIssueEvent(ctx context.Context, event *PullRequestEvent) error {
 	if event == nil || event.Issue == nil {
-		return
+		return errors.New("could not handle issue event: either event or issue is nil")
```

Comment:
I am wondering whether this should be considered as an error scenario?

Reply:
Actually I thought about this, from the function perspective it is an error because it requires these values to be set. Actually we should be sure about these values not to be nil before coming this point. So, the check was here and I deferred this to handler separation.

Reply:
I see. Makes sense.